d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -833,4 +854,59 @@ private ImmutablePair<String, String> getZnodePathAndData(String cluster) {
   public boolean isWarmingUp() {
     return _isWarmingUp;
   }
+
+  private Map<Integer, PartitionData> validatePartitionData(Map<Integer, PartitionData> partitionData) {
```

Comment:
If we make this function package-private instead of fully private, we can probably add some tests for it in `TestZookeeperAnnouncer`, which I feel like might be useful, especially in validating the behavior of the `RECTIFY` action

Reply:
makes sense. added.